jahia-cloud-env-update/update-to-3.2.0/update.yml

```
@@ -14,13 +14,44 @@ onInstall:
       log: Nothing to do
 
 actions:
+  # https://jira.jahia.org/browse/PAAS-1002 changes were never applied to existing environments
+  # For instance Arkema's environments
+  # I prefer adding these values to jahia.properties before "moving" them to envvars in case we have to
+  # revert these changes for some reasons
+  fixMissingEhcacheValues:
+    - cmd [proc]: |-
+        grep -q org.jahia.ehcachemanager $STACK_PATH/conf/digital-factory-config/jahia/jahia.properties || echo $jahia_cfg_operatingMode
+    - set:
+        operatingMode: "${response.out}"
+    - if ("${this.operatingMode}" != ""): # if the ehcachemanager values are not found in jahia.properties
+        - if ("${this.operatingMode}" == "production"):
+            - set:
+                jahia_cfg_org_jahia_ehcachemanager_maxBytesLocalHeap: 800M
+                jahia_cfg_org_jahia_ehcachemanager_big_maxBytesLocalHeap_cp: 2500M
+                jahia_cfg_org_jahia_ehcachemanager_big_maxBytesLocalHeap_proc: 700M
+        - else:
+            - set:
+                jahia_cfg_org_jahia_ehcachemanager_maxBytesLocalHeap: 700M
+                jahia_cfg_org_jahia_ehcachemanager_big_maxBytesLocalHeap_cp: 700M
+                jahia_cfg_org_jahia_ehcachemanager_big_maxBytesLocalHeap_proc: 700M
+        - cmd [proc]: |-
+            echo "org.jahia.ehcachemanager.maxBytesLocalHeap=${this.jahia_cfg_org_jahia_ehcachemanager_maxBytesLocalHeap}" \
+              >> $STACK_PATH/conf/digital-factory-config/jahia/jahia.properties
+            echo "org.jahia.ehcachemanager.big.maxBytesLocalHeap=${this.jahia_cfg_org_jahia_ehcachemanager_big_maxBytesLocalHeap_proc}" \
+              >> $STACK_PATH/conf/digital-factory-config/jahia/jahia.properties
+        - cmd [cp]: |-
+            echo "org.jahia.ehcachemanager.maxBytesLocalHeap=${this.jahia_cfg_org_jahia_ehcachemanager_maxBytesLocalHeap}" \
+              >> $STACK_PATH/conf/digital-factory-config/jahia/jahia.properties
+            echo "org.jahia.ehcachemanager.big.maxBytesLocalHeap=${this.jahia_cfg_org_jahia_ehcachemanager_big_maxBytesLocalHeap_cp}" \
+              >> $STACK_PATH/conf/digital-factory-config/jahia/jahia.properties
+
   jahiaPropertyToEnvVar:
     - cmd [proc]: |-
         PROPERTIES_FILE=/opt/tomcat/conf/digital-factory-config/jahia/jahia.properties
         ENVVAR=${this}
         PROPERTY_KEY="$(echo $ENVVAR | sed "s;jahia_cfg_;;" | sed "s;_;.;g")"
         if ! grep -wq $ENVVAR /.jelenv; then
-          PROPERTY_VALUE=$(grep -E "^[# ]*$PROPERTY_KEY *=" $PROPERTIES_FILE | sed "s;.*= *\(.*\);\1;")
+          PROPERTY_VALUE=$(grep -E "^ *$PROPERTY_KEY *=" $PROPERTIES_FILE | sed "s;.*= *\(.*\);\1;")
```

```suggestion
          PROPERTY_VALUE=$(awk -F'=' -v key='\\<'$PROPERTY_KEY'\\>' '$0~key {if($1~/^#/){commented_value=$NF}; if($1!~/^#/){value=$NF}} END{if(value){gsub(/\s+/, "", value); print value}else{gsub(/\s+/,"",commented_value); print}}' $PROPERTIES_FILE)
```

i will be more confident with something like that in case a we got something like this:
```ini
param1 = hop
#param1 = nothopatall
```
with the `awk` here, we can be sure that it will always be the last none commented line for a parameter that will be return, or the last commented one if they are no uncommented line.


Are we sure that a _naive_ grep will be ok ?

---

I check on *all* environments (prod cluster) with Ansible and the grep works properly.
Actually these values were explicitely defined either in docker image or in universal JPS, so there is absolutely no need to check for commented lines.
So I guess you can merge the PR, pleeaaaaaaase :pleading_face: